chia/full_node/full_node_api.py

```diff
@@ -726,19 +726,29 @@ async def declare_proof_of_space(
             # 2. In the same sub-slot as the peak
             # 3. In a future sub-slot that we already know of
 
-            # Checks that the proof of space is valid
-            quality_string: Optional[bytes32] = verify_and_get_quality_string(
-                request.proof_of_space, self.full_node.constants, cc_challenge_hash, request.challenge_chain_sp
-            )
-            assert quality_string is not None and len(quality_string) == 32
-
             # Grab best transactions from Mempool for given tip target
             aggregate_signature: G2Element = G2Element()
             block_generator: Optional[BlockGenerator] = None
             additions: Optional[List[Coin]] = []
             removals: Optional[List[Coin]] = []
             async with self.full_node.blockchain.priority_mutex.acquire(priority=BlockchainMutexPriority.high):
                 peak: Optional[BlockRecord] = self.full_node.blockchain.get_peak()
+
```

Comment:
its too bad we have to tie up the blockchain mutex to get the proof. i guess this is because we need the height?

Reply:
right. I did this reluctantly. I can't think of a better way to do it. if we don't hold  the mutex `peak` and the peak height may not be constant. However, I don't believe `verify_and_get_quality_string()` does any disk I/O though, it's just computation. e.g. it's not `async`

Reply:
Resolving